src/test/java/org/kohsuke/github/GHWorkflowRunTest.java

```diff
@@ -0,0 +1,173 @@
+package org.kohsuke.github;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Date;
+import java.util.List;
+
+import org.awaitility.Awaitility;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.kohsuke.github.GHWorkflowRun.Conclusion;
+import org.kohsuke.github.GHWorkflowRun.Status;
+
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration;
+
+@Ignore
+public class GHWorkflowRunTest extends AbstractGitHubWireMockTest {
+
+    private static String REPO_NAME = "hub4j-test-org/GHWorkflowRunTest";
+
+    private GHRepository repo;
+
+    private Duration atLeast;
+    private Duration pollInterval;
+    private Duration atMost;
+
+    private long cancelledWorkflowRunId;
+    private long workflowRunIdToDelete;
+
+    @Override
+    protected WireMockConfiguration getWireMockOptions() {
+        return super.getWireMockOptions().extensions(templating.newResponseTransformer());
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        repo = gitHub.getRepository(REPO_NAME);
+
+        if (mockGitHub.isUseProxy()) {
+            atLeast = Duration.ofSeconds(5);
+            pollInterval = Duration.ofSeconds(5);
+            atMost = Duration.ofSeconds(60);
+        } else {
+            atLeast = Duration.ofMillis(1);
+            pollInterval = Duration.ofMillis(1);
+            atMost = Duration.ofMillis(12);
+        }
+    }
+
+    @Test
+    public void testManualRunAndBasicInformation() throws IOException {
+        GHWorkflow workflow = repo.getWorkflow("fast-workflow.yml");
+
+        Date referenceDate = templating.testStartDate;
```

Comment:
Here, I need to have some recording of the date of the test. I thought that maybe this could be useful but apparently not as it's not recorded AFAICS.

Do you have any idea how I could record the start date to be sure I can test things properly?

Reply:
I went for a run and things are clearer now. I think I will just base things on the last id, that will work and be less sensitive to time inconsistencies.

Reply:
I was able to fix it and get the tests to work (hopefully). They are stable on my laptop, let's see on CI.